source/beamline/CLS/CLSBeamlineStatus.cpp

```diff
@@ -40,6 +40,14 @@ bool CLSBeamlineStatus::isOff() const
 	}
 }
 
+AMControl * CLSBeamlineStatus::beamlineStatusPVControl()
+{
+	if (beamlineStatusPVControl_ )
+		return beamlineStatusPVControl_ ;
+	else
+		return this;
```

Comment:
I still don't like it because the behaviour that comes from calling it this way doesn't necessarily match the expected control flow.

I am prepared to merge this code in, but I want to emphasize that programming in this way is not clear and will likely cause misleading behaviour when debugging it in the future.  Personally, I prefer code clarity than simplicity - which should be paramount considering that we spend most of our time debugging old, hard to read code.